src/streamsync/serve.py

```diff
@@ -41,7 +42,9 @@ async def lifespan(app: FastAPI):
         app_runner.hook_to_running_event_loop()
         app_runner.load()
 
-        if on_load is not None:
+        if on_load is not None \
+           and hasattr(app.state, 'is_server_static_mounted') \
+           and app.state.is_server_static_mounted:
```

Comment:
Why are we checking `is_server_static_mounted` here? My idea is that loading without `/static` should work similarly to a normal run, but log the error message. Not sure why we're only calling `on_load()` only if this succeeded. Has the lack of static caused any issues somewhere else?

Reply:
My idea with this addition was that `on_load()`  message (“_App/Builder is available at 127.0.0.1:5000_”) would be misleading in this particular case, because when server static is not mounted, the app is not accessible at 127.0.0.1:5000 (when opened, it displays a JSON object `{"detail": "Not found"}`) – you’d have either to do `npm run build` to add server static and then restart the app, or to launch the UI separately with `npm run dev` as the contributing doc (and the new error message) suggest – but in this case, the frontend becomes accessible through the localhost link that Vite provides you with, and 127.0.0.1:5000 still displays the same JSON I’ve mentioned previously.